### ISSUE TITLE
[MASSEMBLY-870] Regression: Repository downloads all dependencies remotely into _tmp folders

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-assembly-plugin#1080]
**[Tuomas Kiviaho](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=tuomas_kiviaho)** opened **[MASSEMBLY-870](https://issues.apache.org/jira/browse/MASSEMBLY-870?redirect=false)** and commented

Repository `_tmp` folders under `achive-tmp` are filled with dependency artifacts (.jar files and all) from pom which are not met by the filtering. I noticed this when shifting from 2.6 to 3.1.0 and it is quite critical because these dependencies do not come from local repo, but instead they are always downloaded remotely.

I do have `dependency:copy` also before this plugin which doesn't do this so I bet it has something to do with how the Aether is setup in `DefaultRepositoryAssembler`. By removing the `<repositories>` section the assembly stopped this.

The end result is now that everything that I've got in my pom is being filled into `runner/tmp/repo_tmp`.

```
     <repository>
          <scope>provided</scope>
          <includeMetadata>false</includeMetadata>
          <includes>
               <include>org.mortbay.jetty.alpn:alpn-boot</include>
          </includes>
          <outputDirectory>runner/tmp/repo</outputDirectory>
     </repository>
</repositories>
```

Once the assembly is complete the _tmp folder gets deleted and the end result contains just that one jar file.

This is kind of a blocker, because I bet people have got used over the years to cleaning the target folder so that they can get a nice assembly each time a dependency is being upgraded despite the efforts of incremental builds. Due to the cleaning there will always be repeated remote downloads because local repository in settings is not utilized.

---

**Affects:** 3.1.0

**Issue Links:**
- [MASSEMBLY-957](https://issues.apache.org/jira/browse/MASSEMBLY-957) Deprecate the repository element in assembly decriptor

- [MASSEMBLY-974](https://issues.apache.org/jira/browse/MASSEMBLY-974) Drop deprecated repository element

- [MASSEMBLY-874](https://issues.apache.org/jira/browse/MASSEMBLY-874) maven-assembly plugin always downloads dependencies from net

2 votes, 4 watchers

**Comment by jira-importer** [comment]
**[Tuomas Kiviaho](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=tuomas_kiviaho)** commented

This was quite educating discussion http://maven.40175.n5.nabble.com/API-to-resolve-a-version-range-artifact-td5889934.html because https://github.com/apache/maven-plugins/blob/55da60389e9e8326a022bf6b93484978fdb263ed/maven-assembly-plugin/src/main/java/org/apache/maven/plugins/assembly/repository/DefaultRepositoryAssembler.java uses the same approach.

[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte) I tested with my config that my invoker plugin still seems to be playing along nicely. I've been using the Fast Build Configuration https://maven.apache.org/plugins/maven-invoker-plugin/examples/fast-use.html and I think that kind of approach would solve this problem as well.

**Comment by jira-importer** [comment]
**[Tuomas Kiviaho](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=tuomas_kiviaho)** commented

I tried to upgrade from 2.6 to 3.1.1 and the problem still remains. Local repository is still not utilized and the build takes way too long.

**Comment by jira-importer** [comment]
**[Tuomas Kiviaho](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=tuomas_kiviaho)** commented

And on top of that the artifacts that I had installed to local repo are not taken into account

**Comment by jira-importer** [comment]
**[Tamas Cservenak](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=cstamas)** commented

Confirmed, same as MASSEMBLY-874: due repository element, this error is triggered.

**Comment by jira-importer** [comment]
**[Slawomir Jaranowski](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=slawekjaranowski)** commented

As repository element was removed in MASSEMBLY-974 I close as won't fix
